interface/resources/qml/hifi/dialogs/security/EntityScriptQMLWhitelist.qml

```diff
@@ -21,62 +21,94 @@ import "../../../windows"
-    var whitelistText = arrayWhitelist.join("\n");
-    return whitelistText;
-  }
-  
-  function setWhitelistAsText(whitelistText) {
-    Settings.setValue("private/settingsSafeURLS", whitelistText.text);
-    
-    var originalSetString = whitelistText.text;
-    var originalSet = originalSetString.split(' ').join('');
-    
-    var check = Settings.getValue("private/settingsSafeURLS");
-    var arrayCheck = check.split(",");
-    var textCheck = arrayCheck.join("\n");
-    
-    if(textCheck == originalSet) {
-      setWhitelistSuccess(true);
-    } else {
-      setWhitelistSuccess(false);
+    id: parentBody;
+
+    function getWhitelistAsText() {
+        var whitelist = Settings.getValue("private/settingsSafeURLS");
+        var arrayWhitelist = whitelist.split(",").join("\n");
+        return arrayWhitelist;
     }
-  }
-  
-  function setWhitelistSuccess(success) {
-    if(success) {
-      notificationText.text = "Successfully saved settings.";
-    } else {
-      notificationText.text = "Error! Settings not saved.";
+
+    function setWhitelistAsText(whitelistText) {
+        Settings.setValue("private/settingsSafeURLS", whitelistText.text);
+
+        var originalSetString = whitelistText.text;
+        var originalSet = originalSetString.split(' ').join('');
+
+        var check = Settings.getValue("private/settingsSafeURLS");
+        var arrayCheck = check.split(",").join("\n");
+
+        setWhitelistSuccess(arrayCheck === originalSet);
+    }
+
+    function setWhitelistSuccess(success) {
+        if (success) {
+            notificationText.text = "Successfully saved settings.";
+        } else {
+            notificationText.text = "Error! Settings not saved.";
+        }
+    }
+
+    function toggleWhitelist(enabled) {
+        Settings.setValue("private/whitelistEnabled", enabled);
+        console.info("Toggling Whitelist to:", enabled);
+    }
+
+    function initCheckbox() {
+        var check = Settings.getValue("private/whitelistEnabled", false);
+
+        if (check) {
+            whitelistEnabled.toggle();
+        }
     }
-  }
   
   
   anchors.fill: parent
   width: parent.width;
   height: 120;
   color: "#80010203";
 
-  HifiStylesUit.RalewayRegular {
-      id: titleText;
-      text: "Entity Script Whitelist"
-      // Text size
-      size: 24;
-      // Style
-      color: "white";
-      elide: Text.ElideRight;
-      // Anchors
-      anchors.top: parent.top;
-      anchors.left: parent.left;
-      anchors.leftMargin: 20;
-      anchors.right: parent.right;
-      anchors.rightMargin: 20;
-      height: 60;
-  }
+	HifiStylesUit.RalewayRegular {
+		id: titleText;
+		text: "Entity Script / QML Whitelist"
+		// Text size
+		size: 24;
+		// Style
+		color: "white";
+		elide: Text.ElideRight;
+		// Anchors
+		anchors.top: parent.top;
+		anchors.left: parent.left;
+		anchors.leftMargin: 20;
+		anchors.right: parent.right;
+		anchors.rightMargin: 20;
+		height: 60;
```

Comment:
I notice the conflicting usage of mixed spaces and tab all over this file. Please convert to spaces.